activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -714,7 +714,29 @@ def reconnect
         def configure_connection
           @raw_connection.busy_timeout(self.class.type_cast_config_to_integer(@config[:timeout])) if @config[:timeout]
 
+          # Enforce foreign key constraints
+          # https://www.sqlite.org/pragma.html#pragma_foreign_keys
+          # https://www.sqlite.org/foreignkeys.html
           raw_execute("PRAGMA foreign_keys = ON", "SCHEMA")
+          unless @memory_database
+            # Journal mode WAL allows for greater concurrency (many readers + one writer)
```

Comment:
Is there any risk to enabling all of these things on an existing app's SQLite database?

Reply:
Journal modes are switchable without issue. Any harmful switches (like from WAL to DELETE are prevented by the SQLite engine itself: https://sqlite.org/forum/info/0961468fc668f83569fdfb46939fe14ca5bf17b72178f5335b87f2279522f44d)

The only risk is with dropping the durability guarantee. There is no increased corruption risk (unless you are using something ancient like EXT2), but you are now syncing every 1000 page write instead of every page write, some data can be lost in case of a power loss (not process crash, that would be ok) right after writing. But this is simply true of NORMAL synchronous mode, regardless of a new db or switching a db. And NORMAL is the recommended synchronous mode to be used with WAL journaling. 

All in all, configuring new connections to existing databases should have no negative side-effects.